### ISSUE TITLE
Collabora Office / Android does not load whole document

### ISSUE THREAD

**Issue by HelmutL** [CollaboraOnline/online#5808]
**Describe the bug**
Collabora Office Version 21.11.6.1 on Android 10 only loads a part of a writer document.

**To Reproduce**
1. Open the attached document
2. Scroll to the end
3. 
4. See error (seems to end at 18.04.2022 although the document contains data until 02.2023)

**Expected behavior**
load whole document

**Actual behavior**
Only loads a part

**Smartphone (please complete the following information)**
 - Device: Honor View 20
 - OS: Android 10

**Additional context**
[Pinzgau.odt](https://github.com/CollaboraOnline/online/files/10608916/Pinzgau.odt)

**Comment by Ezinnem** [comment]
I can confirm. I tested comparatively with the sample file in Online and also using the Andriod CO office app.

**Comment by Tex2002ans** [comment]
I confirm too.

On Android, CO 21.11.6.1.

Mine reaches a little further down into the sample file.

Makes it to this table:

> Tabelle zuletzt aktualisiert: 28.04.2022 15:31:03

(Page 77/295)

**Comment by Tex2002ans** [comment]
I believe this issue is now solved. :)

@HelmutL Can you retest this on your device using the latest app and see if it fixed it for you too?

- - -

The document loaded fully for me in the latest Android app:

> COOLWSD version: 23.05.4.1
> git hash: b3abc03d
> LOKit version:
> Collabora Office 23.05.4.1
> git hash: 966b9f8

I:

0. Loaded document.
1. Pressed Edit button.
2. In the "Read-Only" popup, I chose to make a copy.

The phone looked like it froze, chugged for a while... but ~60 seconds later, the document eventually popped in.

I was able to scroll all the way to the last page, 295/295.

**Comment by pedropintosilva** [comment]
@Ezinnem

**Comment by timur-g** [comment]
Current Andoroid snapshots are broken, text is not seen, so I would not confirmed this as fixed.

**Comment by timur-g** [comment]
Tested in 24.04 snapshot.
Now, in RO mode I can hardly scroll just 4 pages, further are empty, which is a regression.
In Edit mode, I can scroll till the end.

@Minion3665 please see.

**Comment by Minion3665** [comment]
>  Now, in RO mode I can hardly scroll just 4 pages, further are empty, which is a regression.

I'm able to scroll as far as I want even in read-only mode, but at a point there's a place where it takes a really long time to load past - once it has gotten past that point, scrolling further down is - not "flawless" (some tiles still take a noticeable few moments to load) but far better

---

Because of that, I believe this to be a performance problem

My observation matches quite well with this comment earlier in the thread

> The phone looked like it froze, chugged for a while... but ~60 seconds later, the document eventually popped in.

I'm testing on a snapshot which contains e2ad3cf as its final online commit, and LibreOffice/core@c025051 as its final core commit (the 2024-08-18 snapshot)

**Comment by timur-g** [comment]
I could scroll in RO till the end 02.2023 with 23.05.7.
With the last and same commit snapshot 24.04, as described. Plus all white on zoom change (I think there is a separate report).
It is armeabi-v7a, as previously I could not run aarch64. May it be that is the difference.

**Comment by Minion3665** [comment]
> Plus all white on zoom change (I think there is a separate report).

Yep, that was #9654, which I had believed fixed in https://gerrit.libreoffice.org/c/core/+/171578 - if you're still getting it on a recent snapshot then that surprises me a lot - please make a report or reopen #9654 with further details on your setup/a video/etc.

> [Plus all white on zoom change], [I can hardly scroll just 4 pages, further are empty]

> The phone looked like it froze, chugged for a while... but ~60 seconds later, the document eventually popped in.

I can't emphasise enough how much time this takes. Infact, scrolling really fast on my tablet can overload it to the point where the document crashes out; it's really, really chugging. My phone can handle it, a low-end Android tablet cannot.

I think this is a very severe performance issue, but I think it eventually renders properly. In other words, I think it's much more likely that the tiles haven't loaded in than that the document is really rendering white. It's still very bad, and you may feel that this is a distinction without a difference, but when triaging and fixing it I think the distinction matters quite a bit.

**Comment by timur-g** [comment]
We have Perf tag from before, really scroll was not instant in 23.05, but I keep Regression in 24.04, where it is unusable.

**Comment by Minion3665** [comment]
I'm seeing an extreme amount of spurious invalidations and duplicate tiles being rendered out in my logs (see attached a screenshot of the results of some python code I'm using to find duplicate log lines). I guess everything is being gummed up by having so much rendering and sending of tiles that don't need to be sent at all...

![image](https://github.com/user-attachments/assets/4db45a72-c301-4cdf-bc40-a1e4cee55826)
